include/kamping/p2p/sendrecv.hpp

```diff
@@ -0,0 +1,267 @@
+            {},
+            args...
+        );
+    constexpr auto tag_type = std::remove_reference_t<decltype(recv_tag_param)>::tag_type;
+    if constexpr (tag_type == internal::TagType::value) {
+        int recv_tag = recv_tag_param.tag();
+        KASSERT(
+            Environment<>::is_valid_tag(recv_tag),
+            "invalid recv tag " << recv_tag << ", must be in range [0, " << Environment<>::tag_upper_bound() << "]"
+        );
+    }
+
+    using default_status_param_type = decltype(kamping::status(kamping::ignore<>));
+    auto status =
+        internal::select_parameter_type_or_default<internal::ParameterType::status, default_status_param_type>(
+            {},
+            args...
+        )
+            .construct_buffer_or_rebind();
+
+    using default_source_buf_type = decltype(kamping::source(rank::any));
+    auto&& source_param =
+        internal::select_parameter_type_or_default<internal::ParameterType::source, default_source_buf_type>(
+            {},
+            args...
+        );
+
+    KASSERT(internal::is_valid_rank_in_comm(source_param, *this, true, true));
+    int source = source_param.rank_signed();
+
+    // Calculate rev_count if not given by calling sendrecv with the send_count
+    if constexpr (internal::has_to_be_computed<decltype(recv_count_param)>) {
+        this->sendrecv(
+            kamping::destination(destination.rank_signed()),
+            kamping::send_count(1),
+            kamping::send_buf(send_count.get_single_element()),
+            kamping::recv_count(1),
+            kamping::recv_buf(recv_count_param)
+        );
+    }
+
+    // Ensure that we do not touch the recv buffer if MPI_PROC_NULL is passed,
+    // because this is what the standard guarantees.
+    // Resize the buffer if requested and ensure that it is large enough
+    if constexpr (std::remove_reference_t<decltype(source_param)>::rank_type != internal::RankType::null) {
+        auto compute_required_recv_buf_size = [&] {
+            return asserting_cast<size_t>(recv_count_param.get_single_element());
+        };
+        recv_buf.resize_if_requested(compute_required_recv_buf_size);
+        KASSERT(
+            // if the recv type is user provided, kamping cannot make any assumptions about the required size of the
+            // recv buffer
+            recv_type_is_in_param || recv_buf.size() >= compute_required_recv_buf_size(),
+            "Recv buffer is not large enough to hold all received elements.",
+            assert::light
+        );
+    }
+
+    [[maybe_unused]] int err = MPI_Sendrecv(
+        send_buf.data(),                             // send_buff
+        send_count.get_single_element(),             // send_count
+        send_type.get_single_element(),              // send_data_type
+        destination.rank_signed(),                   // destination
+        send_tag,                                    // send_tag
```

Comment:
```suggestion
        send_tag_param.tag(),                                    // send_tag
```
and remove the `send_tag` variable, it's not used elsewhere. Just to make the code a bit cleaner.